File: fabric-audit-agent/core/narrative.js
const VERDICT_TEXT = {
  'size-up': 'a capacity increase is warranted',
  'optimize': 'optimization opportunities remain before any capacity increase',
  'healthy': 'capacity is healthy',
  'unknown': 'capacity status is unknown',
};

/**
 * Deterministic leadership narrative from an exec view. Pure.
 * (At transfer, the Claude reasoner can rewrite this from the same sanitized inputs.)
 * @param {object} execView  output of viewFor(envelope, 'exec')
 * @returns {string}
 */
export function execNarrative(execView = {}) {
  const v = execView;
  const parts = [
    `Estate health is ${v.health ?? '—'}/100 with ${v.critical ?? 0} critical and ${v.warning ?? 0} warning finding(s).`,
    `On capacity, ${VERDICT_TEXT[v.verdict] ?? 'status is unclear'}.`,
  ];
  if (v.accountability) parts.push(`${v.accountability} issue(s) have been flagged repeatedly without resolution.`);
  if (v.topFindings?.length) parts.push(`Top priority: ${v.topFindings[0].what}`);
  return parts.join(' ');
}

File: fabric-audit-agent-py/fabric_audit_agent/narrative.py
"""Deterministic leadership narrative. Port of ``core/narrative.js``. Pure."""

_VERDICT_TEXT = {
    "size-up": "a capacity increase is warranted",
    "optimize": "optimization opportunities remain before any capacity increase",
    "healthy": "capacity is healthy",
    "unknown": "capacity status is unknown",
}


def exec_narrative(exec_view=None):
    v = exec_view or {}
    health = v.get("health")
    health = health if health is not None else "—"
    critical = v.get("critical")
    critical = critical if critical is not None else 0
    warning = v.get("warning")
    warning = warning if warning is not None else 0
    parts = [
        f"Estate health is {health}/100 with {critical} critical and {warning} warning finding(s).",
        f"On capacity, {_VERDICT_TEXT.get(v.get('verdict'), 'status is unclear')}.",
    ]
    if v.get("accountability"):
        parts.append(f"{v['accountability']} issue(s) have been flagged repeatedly without resolution.")
    if v.get("topFindings"):
        parts.append(f"Top priority: {v['topFindings'][0]['what']}")
    return " ".join(parts)
